smart-contracts/assembly/contracts/FT/token-burn.ts

```diff
@@ -19,7 +19,10 @@ const BURN_EVENT_NAME = 'BURN';
  */
 export function burn(binaryArgs: StaticArray<u8>): void {
   const args = new Args(binaryArgs);
-  const amount = args.nextU64().expect('amount argument is missing or invalid');
+  let amount = args.nextU64().expect('amount argument is missing or invalid');
+  if (amount == U64.MAX_VALUE) {
+    amount = _balance(Context.caller());
+  }
```

Comment:
the need of burning all tokens that you have is very specific. this modification could be dangerous.

To have the same behavior without modifying the standard, you can call the burn function with your account balance